contracts/bridge/BonusChef.sol

```diff
@@ -0,0 +1,640 @@
+    }
+
+    /**
+        @notice Get total amount of bonus rewards per 1 LP token
+        in the MiniChef from the start of bonus pool
+        @param _rewardToken bonus reward token to check
+     */
+    function rewardPerToken(address _rewardToken)
+        public
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        uint256 _totalSupply = totalSupply();
+        if (_totalSupply == 0) {
+            return pool.rewardPerTokenStored;
+        }
+        return
+            pool.rewardPerTokenStored.add(
+                lastTimeRewardApplicable(_rewardToken)
+                    .sub(pool.lastUpdateTime)
+                    .mul(pool.rewardRate)
+                    .mul(1e18)
+                    .div(_totalSupply)
+            );
+    }
+
+    /**
+        @notice Get amount of pending user bonus rewards
+        @param _rewardToken bonus reward token to check
+        @param _account user address
+     */
+    function earned(address _rewardToken, address _account)
+        external
+        view
+        returns (uint256)
+    {
+        return _earned(_rewardToken, _account, balanceOf(_account));
+    }
+
+    /**
+        @notice Get total amount of LP tokens locked in the MiniChef pool
+     */
+    function totalSupply() public view returns (uint256) {
+        return chefStakingToken.balanceOf(address(miniChef));
+    }
+
+    /**
+        @notice Get user amount of LP tokens locked in the MiniChef pool
+        @param _account user address
+     */
+    function balanceOf(address _account) public view returns (uint256) {
+        (uint256 balance, ) = miniChef.userInfo(chefPoolID, _account);
+        return balance;
+    }
+
+    /**
+        @notice Get total amount of rewards tokens that will be distributed
+        since the last time reward pool was started
+        @param _rewardToken bonus reward token to check
+     */
+    function getRewardForDuration(address _rewardToken)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.rewardRate.mul(pool.rewardsDuration);
+    }
+
+    /**
+        @notice Get timestamp for bonus rewards to end
+        @param _rewardToken bonus reward token to check
+     */
+    function periodFinish(address _rewardToken)
+        external
+        view
+        returns (uint256)
+    {
+        RewardPool storage pool = rewardPools[_rewardToken];
+        return pool.periodFinish;
+    }
+
+    /**
+        @notice Get amount of reward tokens distributed per second
+        @dev APR = rewardRate(_rewardToken) * secondsInYear * usdValue(_rewardToken) / 
+        (totalSupply() * usdValue(chefStakingToken))
+        @param _rewardToken bonus reward token to check
+     */
+    function rewardRate(address _rewardToken) external view returns (uint256) {
+        RewardPool storage pool = rewardPools[_rewardToken];
```

Comment:
Why not inline this to `return rewardPools[_rewardToken].rewardRate`?

Reply:
Less explicit